### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by gregorsart**
Last PR for the app maybe 💯 

If someone has an idea on how to change the color of the `ImageTextModule` depending on two things:

1.  The passed react props (should there be a change of color yes or no)
2. The active theme mode (dark / light) we get via react context api

And last but not least we "invented" a working solution for the `CreateGlobalStyles` and implemented it in app.js in `line 23`

`{themeMode === "light" ? <GlobalStyle /> : <DarkModeStyles />}`

We are not sure if this is best practice (but it works)

### CODE REVIEW

File: components/DarkmodeButton.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+              type="checkbox"
+              value=""
+              checked={themeMode === "dark"}
+              className=""
+              onChange={onChangeToDark}
+            />
+          </Toggle>
+        </ToggleBorder>
+      </ToggleContainer>
+    </>
```
Comment (ThomasKirst) [inline components/DarkmodeButton.js:34]: No fragment needed here:

```suggestion
      <ToggleContainer>
        <ToggleBorder>
          <Toggle $isToggled={isToggled}>
            <StyledInput
              type="checkbox"
              value=""
              checked={themeMode === "dark"}
              className=""
              onChange={onChangeToDark}
            />
          </Toggle>
        </ToggleBorder>
      </ToggleContainer>
```

File: styles.js [hunk 2]
```diff
@@ -16,6 +16,66 @@ export const basicFont = Poppins({
...
+    --basicmargin: 2.4rem;
+  }
+
+  p {
+    line-height: 1.6;
+    font-size: 1rem;
+    color: var(--font-color);
+    font-weight: 100;
+   
+
```
Comment (ThomasKirst) [inline styles.js:69]: Please prettify (remove double empty lines).

File: styles.js [hunk 3]
```diff
@@ -16,6 +16,66 @@ export const basicFont = Poppins({
...
+    max-width: 100%;
+  }
+
+  input, button, textarea, select {
+    font: inherit;
+  }
+
+  p, h1, h2, h3, h4, h5, h6 {
+    overflow-wrap: break-word;
+  }
```
Comment (ThomasKirst) [inline styles.js:40]: The styles which are shared both by light and dark mode could be written in another variable and then included here as `${sharedGlobalStyles}` or similar.

File: components/DarkmodeButton.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      <ToggleContainer>
+        <ToggleBorder>
+          <Toggle $isToggled={isToggled}>
+            <StyledInput
+              type="checkbox"
+              value=""
+              checked={themeMode === "dark"}
+              className=""
+              onChange={onChangeToDark}
+            />
```
Comment (ThomasKirst) [inline components/DarkmodeButton.js:30]:
```suggestion
            <StyledInput
              type="checkbox"
              checked={themeMode === "dark"}
              onChange={onChangeToDark}
            />
```

File: components/Header.js [hunk 5]
```diff
@@ -20,35 +21,55 @@ export default function Header() {
...
+        <StyledWrapper>
+          <DarkmodeButton />
+          <StyledHamburgerButton type="button" onClick={handleNavigation}>
+            <StyledHamburgerIcon
+              src="/hamburger-icon.svg"
+              alt="Icon of a hamburger menu"
+              width={35}
+              height={35}
+              blurDataURL="data:..."
+              placeholder="blur" // Optional blur-up while loading
```
Comment (ThomasKirst) [inline components/Header.js:34]: Does this serve any purpose now or can it be removed?

File: pages/_app.js [hunk 6]
```diff
@@ -1,11 +1,26 @@
 import GlobalStyle from "../styles";
 import { SWRConfig } from "swr";
+import useLocalStorageState from "use-local-storage-state";
+import { ThemeProvider } from "@/contexts/theme";
 import Layout from "@/components/Layout";
+import { DarkModeStyles } from "../styles";
 
 export default function App({ Component, pageProps }) {
+  const [themeMode, setThemeMode] = useLocalStorageState("themeMode", {
+    defaultValue: ["light"],
```
Comment (ThomasKirst) [inline pages/_app.js:10]: You initially set the `defaultValue` to an array with one item `light`, but I guess you simply want to store the string `light` here.

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
Hi Gregor,

looks pretty amazing to me, well done!

About the question in your PR description: I´m not sure what you want to achieve – maybe open a question on the discussion with a bit more details / context?

About your "invention" of having a separate configuration of Global Styles for the dark mode: I like your solution and I would not overcomplicate it. I left you there a tiny comment on how to improve it a little, but overall it´s a smart solution. You basically just want to override the CSS variables on the given context, that´s great!

There are other ways of "overriding" styles as described [in this article](https://dev.to/britotiagos/using-props-on-styled-component-and-nextjs-51lk), for instance. Then you would have to pass the theme prop to GlobalStyles, but I don´t think that´s necessary.

I hope this helps.
– Thomas

**Review by NikRoe — APPROVED** [review]
LGTM 🙌

**Pull request merged**